pygeos/test/test_creation.py

```diff
@@ -208,3 +208,31 @@ def test_box_multiple():
     actual = pygeos.box(0, 0, [1, 2], [1, 2])
     assert str(actual[0]) == "POLYGON ((1 0, 1 1, 0 1, 0 0, 1 0))"
     assert str(actual[1]) == "POLYGON ((2 0, 2 2, 0 2, 0 0, 2 0))"
+
+
+class BaseGeometry(pygeos.Geometry):
+    @property
+    def type_id(self):
+        return pygeos.get_type_id(self)
+
+
+class Point(BaseGeometry):
+    @property
+    def x(self):
+        return pygeos.get_x(self)
+
+    @property
+    def y(self):
+        return pygeos.get_y(self)
+
+
+def test_subclasses():
+    orig = pygeos.lib.registry[0]
+    pygeos.lib.registry[0] = Point
+
+    for point in [Point("POINT (1 1)"), pygeos.points(1, 1)]:
+        assert isinstance(point, Point)
+        assert pygeos.get_type_id(point) == pygeos.GeometryType.POINT
+        assert point.x == 1
+
+    pygeos.lib.registry[0] = orig
```

Comment:
This teardown should also happen if the test fails. Otherwise we may get confusing situations in the future.

Maybe with a fixture:

```
@pytest.fixture
def with_point_in_registry():
    pygeos.lib.registry[0] = Point
    yield
    pygeos.lib.registry[0] = orig
```

Reply:
Used that fixture!